src/main/angular/app/services/fake_services/fake_redirect.service.ts

```diff
@@ -0,0 +1,22 @@
+import {Injectable} from '@angular/core';
+import {RedirectService} from '../redirect.service';
+
+/** Service used to fake out redirects in the application. */
+@Injectable()
+export class FakeRedirectService implements RedirectService {
```

Comment:
Although in the code that you've used so far, fakes have been very useful because they've also accelerated development, I think this class is unnecessary. 

Instead of implementing a Fake for the purposes of testing interactions, you can use a Spy(https://jasmine.github.io/api/edge/Spy.html). Could you take a look?

Reply:
I looked into spies, and while they are indeed useful they aren't what I want since I'd need to create a mock of the Angular router.

Instead, I found [ts-mockito](https://github.com/NagRock/ts-mockito) which I think is more what you were looking for?